src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -116,6 +116,43 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
                 throw new InvalidOperationException();
             }
 
+            if (_lspSolution.Value.Document is null)
+            {
+                return null;
+            }
+
+            if (_lspSolution.Value.Document is Document document)
+            {
+                return document;
+            }
+
+            // Explicitly throw for attempts to get a Document when only a TextDocument is available.
+            throw new InvalidOperationException();
```

Comment:
We should put a message here so its distinguishable from the other invalid operaiton when the workspace is null